File: mobile/ios/OnlyCoffee/Views/Coupons/CouponCard.swift
import SwiftUI

struct CouponCard: View {
    let coupon: Coupon
    @State private var showDetails = false

    var body: some View {
        Button(action: {
            showDetails = true
        }) {
            VStack(alignment: .leading, spacing: 12) {
                // Header with value badge
                HStack(alignment: .top, spacing: 12) {
                    // Value Badge
                    Text(coupon.displayValue)
                        .font(.system(size: 24, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 16)
                        .padding(.vertical, 12)
                        .background(
                            coupon.isExpired ? Color.gray :
                            coupon.isExpiringSoon ? Color.orange : Color.brandPink
                        )
                        .cornerRadius(12)

                    VStack(alignment: .leading, spacing: 4) {
                        Text(coupon.label)
                            .font(.headline)
                            .fontWeight(.bold)
                            .foregroundColor(coupon.isExpired ? .secondary : .primary)

                        if let description = coupon.description {
                            Text(description)
                                .font(.caption)
                                .foregroundColor(.secondary)
                                .lineLimit(2)
                        }
                    }

                    Spacer()
                }

                // Expiry and Channel Info
                HStack(spacing: 16) {
                    // Expiry
                    HStack(spacing: 4) {
                        Image(systemName: coupon.isExpiringSoon ? "exclamationmark.triangle.fill" : "clock")
                            .font(.caption)
                            .foregroundColor(coupon.isExpiringSoon ? .orange : .secondary)

                        Text(coupon.expiryText)
                            .font(.caption)
                            .fontWeight(coupon.isExpiringSoon ? .semibold : .regular)
                            .foregroundColor(coupon.isExpiringSoon ? .orange : .secondary)
                    }

                    Spacer()

                    // Channel Badge
                    HStack(spacing: 4) {
                        Image(systemName: coupon.channels == "app_only" ? "iphone" : "bag")
                            .font(.caption2)

                        Text(coupon.channelText)
                            .font(.caption2)
                            .fontWeight(.medium)
                    }
                    .foregroundColor(.white)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 4)
                    .background(Color.brandAccent.opacity(0.8))
                    .cornerRadius(8)
                }

                // Use Now Button (only for active coupons)
                if coupon.isActive {
                    Button(action: {
                        // TODO: Navigate to menu or checkout
                    }) {
                        Text("Use Now")
                            .font(.headline)
                            .fontWeight(.semibold)
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 12)
                            .background(Color.brandPink)
                            .cornerRadius(8)
                    }
                } else if coupon.status == .redeemed {
                    HStack {
                        Image(systemName: "checkmark.circle.fill")
                            .foregroundColor(.green)

                        Text("Redeemed")
                            .font(.subheadline)
                            .fontWeight(.medium)
                            .foregroundColor(.green)

                        if let redeemedAt = coupon.redeemedAt {
                            Text("• \(formatDate(redeemedAt))")
                                .font(.caption)
                                .foregroundColor(.secondary)
                        }
                    }
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 8)
                    .background(Color.green.opacity(0.1))
                    .cornerRadius(8)
                }
            }
            .padding()
            .background(Color.white)
            .cornerRadius(16)
            .shadow(color: .black.opacity(0.05), radius: 8, x: 0, y: 2)
            .opacity(coupon.isExpired ? 0.6 : 1.0)
        }
        .buttonStyle(PlainButtonStyle())
        .sheet(isPresented: $showDetails) {
            CouponDetailView(coupon: coupon)
        }
    }

    private func formatDate(_ date: Date) -> String {
        let formatter = DateFormatter()
        formatter.dateFormat = "MMM d"
        return formatter.string(from: date)
    }
}

// MARK: - Coupon Detail View
struct CouponDetailView: View {
    let coupon: Coupon
    @Environment(\.presentationMode) var presentationMode
    @State private var showQRCode = false

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 24) {
                    // Value Display
                    VStack(spacing: 8) {
                        Text(coupon.displayValue)
                            .font(.system(size: 48, weight: .bold))
                            .foregroundColor(.brandPink)

                        Text(coupon.label)
                            .font(.title2)
                            .fontWeight(.bold)

                        if let description = coupon.description {
                            Text(description)
                                .font(.body)
                                .foregroundColor(.secondary)
                                .multilineTextAlignment(.center)
                        }
                    }
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 32)
                    .background(Color.brandLight.opacity(0.2))
                    .cornerRadius(16)

                    // Details
                    VStack(alignment: .leading, spacing: 16) {
                        DetailRow(icon: "clock", title: "Expires", value: coupon.formattedExpiryDate)

                        DetailRow(icon: "bag", title: "Redeem", value: coupon.channelText)

                        if let eligibleItems = coupon.eligibleItems {
                            if let exclude = eligibleItems.exclude, !exclude.isEmpty {
                                DetailRow(
                                    icon: "xmark.circle",
                                    title: "Excludes",
                                    value: exclude.joined(separator: ", ").capitalized
                                )
                            }

                            if let include = eligibleItems.include, !include.isEmpty {
                                DetailRow(
                                    icon: "checkmark.circle",
                                    title: "Includes",
                                    value: include.joined(separator: ", ").capitalized
                                )
                            }
                        }

                        DetailRow(icon: "tag", title: "Source", value: formatSource(coupon.source))
                    }
                    .padding()
                    .background(Color.white)
                    .cornerRadius(16)

                    // Action Buttons
                    if coupon.isActive {
                        VStack(spacing: 12) {
                            // Show QR Code button (for in-store redemption)
                            if coupon.channels == "in_store" || coupon.channels == "both" {
                                Button(action: {
                                    showQRCode = true
                                }) {
                                    HStack {
                                        Image(systemName: "qrcode")
                                        Text("Show QR Code for In-Store")
                                            .fontWeight(.semibold)
                                    }
                                    .font(.headline)
                                    .foregroundColor(.white)
                                    .frame(maxWidth: .infinity)
                                    .padding(.vertical, 16)
                                    .background(Color.green)
                                    .cornerRadius(12)
                                }
                            }

                            // Use in app button
                            if coupon.channels == "app_only" || coupon.channels == "both" {
                                Button(action: {
                                    // TODO: Navigate to menu with coupon pre-selected
                                    presentationMode.wrappedValue.dismiss()
                                }) {
                                    HStack {
                                        Image(systemName: "iphone")
                                        Text("Use in App")
                                            .fontWeight(.semibold)
                                    }
                                    .font(.headline)
                                    .foregroundColor(.white)
                                    .frame(maxWidth: .infinity)
                                    .padding(.vertical, 16)
                                    .background(Color.brandPink)
                                    .cornerRadius(12)
                                }
                            }
                        }
                    }
                }
                .padding()
            }
            .background(Color(.systemGroupedBackground))
            .navigationTitle("Coupon Details")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Done") {
                        presentationMode.wrappedValue.dismiss()
                    }
                }
            }
            .sheet(isPresented: $showQRCode) {
                CouponQRCodeView(coupon: coupon)
            }
        }
    }

    private func formatSource(_ source: String) -> String {
        source.replacingOccurrences(of: "_", with: " ").capitalized
    }
}

// MARK: - Detail Row
struct DetailRow: View {
    let icon: String
    let title: String
    let value: String

    var body: some View {
        HStack(spacing: 12) {
            Image(systemName: icon)
                .font(.title3)
                .foregroundColor(.brandPink)
                .frame(width: 24)

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.caption)
                    .foregroundColor(.secondary)

                Text(value)
                    .font(.body)
                    .foregroundColor(.primary)
            }

            Spacer()
        }
    }
}

#Preview {
    CouponCard(coupon: Coupon(
        id: "1",
        userId: "user1",
        promoCodeId: nil,
        type: .percentOff,
        label: "50% Off Drink",
        description: "Get 50% off any drink",
        valueCents: nil,
        percentOff: 50,
        priceOverrideCents: nil,
        eligibleItems: Coupon.EligibleItems(exclude: ["waffolino"], include: nil),
        channels: "both",
        expiresAt: Date().addingTimeInterval(86400 * 3),
        redeemedAt: nil,
        redeemedOrderId: nil,
        status: .active,
        source: "new_user",
        metadata: nil,
        createdAt: Date()
    ))
    .padding()
}

File: android/app/src/main/java/com/onlycoffee/app/ui/screens/coupons/CouponCard.kt
package com.onlycoffee.app.ui.screens.coupons

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.onlycoffee.app.data.model.Coupon
import com.onlycoffee.app.data.model.CouponStatus
import java.text.SimpleDateFormat
import java.util.*

@Composable
fun CouponCard(
    coupon: Coupon,
    modifier: Modifier = Modifier
) {
    var showDetails by remember { mutableStateOf(false) }
    var showQRCode by remember { mutableStateOf(false) }

    Card(
        modifier = modifier
            .fillMaxWidth()
            .alpha(if (coupon.isExpired) 0.6f else 1.0f),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surface
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        ),
        onClick = { showDetails = true }
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            // Header with value badge
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                verticalAlignment = Alignment.Top
            ) {
                // Value Badge
                Surface(
                    color = when {
                        coupon.isExpired -> Color.Gray
                        coupon.isExpiringSoon -> Color(0xFFFF9800)
                        else -> Color(0xFFE91E63)
                    },
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Text(
                        text = coupon.displayValue,
                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
                        style = MaterialTheme.typography.headlineMedium,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )
                }

                // Label and Description
                Column(
                    modifier = Modifier.weight(1f)
                ) {
                    Text(
                        text = coupon.label,
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = if (coupon.isExpired) {
                            MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                        } else {
                            MaterialTheme.colorScheme.onSurface
                        }
                    )

                    coupon.description?.let { desc ->
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            text = desc,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                            maxLines = 2
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Expiry and Channel Info
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                // Expiry
                Row(
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = if (coupon.isExpiringSoon) {
                            Icons.Default.Warning
                        } else {
                            Icons.Default.Schedule
                        },
                        contentDescription = null,
                        tint = if (coupon.isExpiringSoon) {
                            Color(0xFFFF9800)
                        } else {
                            MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                        },
                        modifier = Modifier.size(16.dp)
                    )

                    Text(
                        text = coupon.expiryText,
                        style = MaterialTheme.typography.bodySmall,
                        fontWeight = if (coupon.isExpiringSoon) FontWeight.SemiBold else FontWeight.Normal,
                        color = if (coupon.isExpiringSoon) {
                            Color(0xFFFF9800)
                        } else {
                            MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                        }
                    )
                }

                Spacer(modifier = Modifier.weight(1f))

                // Channel Badge
                Surface(
                    color = Color(0xFF4CAF50).copy(alpha = 0.8f),
                    shape = RoundedCornerShape(8.dp)
                ) {
                    Row(
                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
                        horizontalArrangement = Arrangement.spacedBy(4.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = if (coupon.channels == "app_only") {
                                Icons.Default.PhoneAndroid
                            } else {
                                Icons.Default.ShoppingBag
                            },
                            contentDescription = null,
                            tint = Color.White,
                            modifier = Modifier.size(12.dp)
                        )

                        Text(
                            text = coupon.channelText,
                            style = MaterialTheme.typography.labelSmall,
                            fontWeight = FontWeight.Medium,
                            color = Color.White
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // Action Buttons
            when {
                coupon.isActive -> {
                    when (coupon.channels) {
                        "in_store", "store_only" -> {
                            // In-store only - show QR code button
                            Button(
                                onClick = { showQRCode = true },
                                modifier = Modifier.fillMaxWidth(),
                                colors = ButtonDefaults.buttonColors(
                                    containerColor = Color(0xFF4CAF50)
                                ),
                                shape = RoundedCornerShape(8.dp)
                            ) {
                                Icon(
                                    imageVector = Icons.Default.QrCode,
                                    contentDescription = null,
                                    modifier = Modifier.size(20.dp)
                                )
                                Spacer(modifier = Modifier.width(8.dp))
                                Text(
                                    text = "Show QR Code for In-Store",
                                    fontWeight = FontWeight.SemiBold
                                )
                            }
                        }
                        "app_only" -> {
                            // App only - show use in app button
                            Button(
                                onClick = { /* TODO: Navigate to menu with coupon pre-selected */ },
                                modifier = Modifier.fillMaxWidth(),
                                colors = ButtonDefaults.buttonColors(
                                    containerColor = Color(0xFFE91E63)
                                ),
                                shape = RoundedCornerShape(8.dp)
                            ) {
                                Text(
                                    text = "Use in App",
                                    fontWeight = FontWeight.SemiBold
                                )
                            }
                        }
                        else -> {
                            // Both channels - show two buttons
                            Column(
                                verticalArrangement = Arrangement.spacedBy(8.dp)
                            ) {
                                Button(
                                    onClick = { showQRCode = true },
                                    modifier = Modifier.fillMaxWidth(),
                                    colors = ButtonDefaults.buttonColors(
                                        containerColor = Color(0xFF4CAF50)
                                    ),
                                    shape = RoundedCornerShape(8.dp)
                                ) {
                                    Icon(
                                        imageVector = Icons.Default.QrCode,
                                        contentDescription = null,
                                        modifier = Modifier.size(20.dp)
                                    )
                                    Spacer(modifier = Modifier.width(8.dp))
                                    Text(
                                        text = "Show QR Code for In-Store",
                                        fontWeight = FontWeight.SemiBold
                                    )
                                }

                                OutlinedButton(
                                    onClick = { /* TODO: Navigate to menu with coupon pre-selected */ },
                                    modifier = Modifier.fillMaxWidth(),
                                    colors = ButtonDefaults.outlinedButtonColors(
                                        contentColor = Color(0xFFE91E63)
                                    ),
                                    shape = RoundedCornerShape(8.dp)
                                ) {
                                    Text(
                                        text = "Use in App",
                                        fontWeight = FontWeight.SemiBold
                                    )
                                }
                            }
                        }
                    }
                }

                coupon.status == CouponStatus.REDEEMED -> {
                    Surface(
                        modifier = Modifier.fillMaxWidth(),
                        color = Color(0xFF4CAF50).copy(alpha = 0.1f),
                        shape = RoundedCornerShape(8.dp)
                    ) {
                        Row(
                            modifier = Modifier.padding(8.dp),
                            horizontalArrangement = Arrangement.spacedBy(8.dp),
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Icon(
                                imageVector = Icons.Default.CheckCircle,
                                contentDescription = null,
                                tint = Color(0xFF4CAF50)
                            )

                            Text(
                                text = "Redeemed",
                                style = MaterialTheme.typography.bodyMedium,
                                fontWeight = FontWeight.Medium,
                                color = Color(0xFF4CAF50)
                            )

                            coupon.redeemedAt?.let { date ->
                                Text(
                                    text = "• ${formatDate(date)}",
                                    style = MaterialTheme.typography.bodySmall,
                                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                                )
                            }
                        }
                    }
                }
            }
        }
    }

    // Coupon Detail Dialog
    if (showDetails) {
        CouponDetailDialog(
            coupon = coupon,
            onDismiss = { showDetails = false }
        )
    }

    // QR Code Screen
    if (showQRCode) {
        CouponQRCodeScreen(
            coupon = coupon,
            onDismiss = { showQRCode = false }
        )
    }
}

@Composable
fun CouponDetailDialog(
    coupon: Coupon,
    onDismiss: () -> Unit
) {
    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Text("Coupon Details")
        },
        text = {
            Column(
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                // Value Display
                Surface(
                    color = Color(0xFFE91E63).copy(alpha = 0.1f),
                    shape = RoundedCornerShape(16.dp)
                ) {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(24.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Text(
                            text = coupon.displayValue,
                            style = MaterialTheme.typography.displayMedium,
                            fontWeight = FontWeight.Bold,
                            color = Color(0xFFE91E63)
                        )

                        Spacer(modifier = Modifier.height(8.dp))

                        Text(
                            text = coupon.label,
                            style = MaterialTheme.typography.titleLarge,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface
                        )

                        coupon.description?.let { desc ->
                            Spacer(modifier = Modifier.height(4.dp))
                            Text(
                                text = desc,
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                            )
                        }
                    }
                }

                // Details
                DetailRow(
                    icon = Icons.Default.Schedule,
                    label = "Expires",
                    value = coupon.formattedExpiryDate
                )

                DetailRow(
                    icon = Icons.Default.ShoppingBag,
                    label = "Redeem",
                    value = coupon.channelText
                )

                coupon.eligibleItems?.let { eligibleItems ->
                    eligibleItems.exclude?.takeIf { it.isNotEmpty() }?.let { excludeList ->
                        DetailRow(
                            icon = Icons.Default.Close,
                            label = "Excludes",
                            value = excludeList.joinToString(", ").capitalize(Locale.getDefault())
                        )
                    }

                    eligibleItems.include?.takeIf { it.isNotEmpty() }?.let { includeList ->
                        DetailRow(
                            icon = Icons.Default.Check,
                            label = "Includes",
                            value = includeList.joinToString(", ").capitalize(Locale.getDefault())
                        )
                    }
                }

                DetailRow(
                    icon = Icons.Default.Label,
                    label = "Source",
                    value = formatSource(coupon.source)
                )
            }
        },
        confirmButton = {
            if (coupon.isActive) {
                TextButton(
                    onClick = {
                        // TODO: Navigate to menu with coupon pre-selected
                        onDismiss()
                    }
                ) {
                    Text("Use This Coupon")
                }
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text("Close")
            }
        }
    )
}

@Composable
fun DetailRow(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    label: String,
    value: String
) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalAlignment = Alignment.Top
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = Color(0xFFE91E63),
            modifier = Modifier.size(24.dp)
        )

        Column(
            verticalArrangement = Arrangement.spacedBy(2.dp)
        ) {
            Text(
                text = label,
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
            )

            Text(
                text = value,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurface
            )
        }
    }
}

private fun formatDate(date: Date): String {
    val format = SimpleDateFormat("MMM d", Locale.getDefault())
    return format.format(date)
}

private fun formatSource(source: String): String {
    return source.replace("_", " ")
        .split(" ")
        .joinToString(" ") { it.capitalize(Locale.getDefault()) }
}
